dist/rules/no-internal.js

```diff
@@ -168,36 +168,40 @@ module.exports = {
         if (jsDoc.tags) {
           for (const tag of jsDoc.tags) {
             if (bannedTags.includes(tag.tagName.escapedText) && isCheckedFile(declaration)) {
-              let name;
-              if (declaration.kind === ts.SyntaxKind.Constructor)
-                name = declaration.parent.symbol.escapedName;
-              else {
-                name = declaration.symbol.escapedName;
-                const parentSymbol = getParentSymbolName(declaration);
-                if (parentSymbol)
-                  name = `${parentSymbol}.${name}`;
-              }
-
-              context.report({
-                node,
-                messageId: "forbidden",
-                data: {
-                  kind: syntaxKindFriendlyNames.hasOwnProperty(declaration.kind) ? syntaxKindFriendlyNames[declaration.kind] : "unknown object type " + declaration.kind,
-                  name,
-                  tag: tag.tagName.escapedText,
+              //Violation key to track and report violations on a per-usage basis
+              const violationKey = `${declaration.kind}_${declaration.symbol.escapedName}_${tag}_${node.range[0]}`;
+              if (!reportedViolationsSet.has(violationKey)) {
```

Comment:
nit: you'd have a smaller diff and imo nicer less-nested code if you `continue`'d on the negative condition rather than use an `if` on the positive condition:
```ts
if (reportedViolationsSet.has(violationKey))
  continue;
  
// remaining code continues without the need for a new {} block
```

Reply:
(you won't have a smaller diff but would have better code if you also do the same thing to the previous `if`)

Reply:
Sounds good, will push the changes